Userland/Applications/PixelPaint/Tools/EraseTool.cpp

```diff
@@ -144,12 +144,39 @@ NonnullRefPtr<Gfx::Bitmap> EraseTool::build_cursor()
     m_scale_last_created_cursor = m_editor ? m_editor->scale() : 1;
     int scaled_size = size() * m_scale_last_created_cursor;
 
+    // If we have an ImageEditor the scaled_size should not excede the longest size of the ImageEditor (width or height)
+    if (m_editor) {
+        int max_dimension = max(m_editor->width(), m_editor->height());
+        if (scaled_size > max_dimension * 2) {
+            // scaled_size needs to change in order to draw the cursor in the correct location.
+            scaled_size = max_dimension * 2;
+        }
+    }
+    scaled_size = max(scaled_size, 1);
+    dbgln("size: {}    scaled_size: {}", size(), scaled_size);
     NonnullRefPtr<Gfx::Bitmap> new_cursor = Gfx::Bitmap::try_create(Gfx::BitmapFormat::BGRA8888, Gfx::IntSize(scaled_size, scaled_size)).release_value_but_fixme_should_propagate_errors();
 
     Gfx::IntRect rect { 0, 0, scaled_size, scaled_size };
     Gfx::Painter painter { new_cursor };
 
-    painter.draw_rect(rect, Color::LightGray);
+    // If no ImageEditor is present, we cannot bind the rect within the editor. Then we will just draw the rect as normal.
+    if (!m_editor) {
+        painter.draw_rect(rect, Color::LightGray);
+    } else {
+        if (m_current_tool_position.x() >= (scaled_size / 2) && (m_current_tool_position.x() + scaled_size / 2) <= m_editor->width() && m_current_tool_position.y() >= (scaled_size / 2) && (m_current_tool_position.y() + scaled_size / 2) <= m_editor->height()) {
```

Comment:
Make `(scaled_size / 2)` a variable, or create a helper function to compute this condition.

Reply:
Moved to external variable to avoid recalculation.